### PR TITLE
Adding dark and light theme names to the sponsor request (params to API)

### PR DESCRIPTION
**Pull request by blckmn**
I couldn't see any way to identify whether or not the dark theme was enabled or not. So added the property to DarkTheme.

If anyway has a better idea please let me know.

### CODE REVIEW

**Comment by haslinghuis** [comment]
@asizon

**Comment by haslinghuis** [comment]
We have something like:

https://github.com/betaflight/betaflight-configurator/blob/ff4b62d7e2f8e3997ad3e704767cff52db3c9229/src/js/tabs/receiver.js#L563-L565

**Comment by asizon** [comment]
> We have something like:
> 
> https://github.com/betaflight/betaflight-configurator/blob/ff4b62d7e2f8e3997ad3e704767cff52db3c9229/src/js/tabs/receiver.js#L563-L565
> 
> and
> 
> https://github.com/betaflight/betaflight-configurator/blob/ff4b62d7e2f8e3997ad3e704767cff52db3c9229/src/js/tabs/options.js#L180-L189

Yes,i think is better to go with the current available function.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/BuildApi.js
Comment (haslinghuis) [inline src/js/BuildApi.js:176]: Checking for `window.matchMedia('(prefers-color-scheme: dark)').matches` would do

File: src/js/DarkTheme.js
Comment (haslinghuis) [inline src/js/DarkTheme.js:12]: This would duplicate configEnabled

Comment (blckmn) [reply src/js/DarkTheme.js:12]: Actually I'm not sure it does. It looked to me that configEnabled is not binary, so supports the option 'honour the system' so can't be relied upon for stating that dark theme was in fact enabled.

Comment (haslinghuis) [reply src/js/DarkTheme.js:12]: Above statement would extract the value or get ConfigStorage `darkTheme` value.

File: src/js/BuildApi.js
Comment (blckmn) [reply src/js/BuildApi.js:176]: I'll test this.

File: src/js/DarkTheme.js
Comment (blckmn) [reply src/js/DarkTheme.js:12]: Which means configEnabled is not well named. It should be configSetting or similar. It can't be relied upon as to if it's actually applied or not... sure it'll have a value of two... but does that mean dark theme is active? I don't think it does because the test of isEnabled is not persisted...

**Comment by blckmn** [comment]
> We have something like:
> 
> https://github.com/betaflight/betaflight-configurator/blob/ff4b62d7e2f8e3997ad3e704767cff52db3c9229/src/js/tabs/receiver.js#L563-L565
> 
> and
> 
> https://github.com/betaflight/betaflight-configurator/blob/ff4b62d7e2f8e3997ad3e704767cff52db3c9229/src/js/tabs/options.js#L180-L189

I actually don't like this approach. The reason is both receiver.js and darktheme.js are coupled together. It would be better that receiver (and any others) registers its interest (observes) the enabled / disabled aspect of darktheme. Then darktheme is entirely standalone... and not coupled to any tabs.

**Comment by haslinghuis** [comment]
Agree, it's part of code from 2020 to fix dark theme on Android.

**Comment by blckmn** [comment]
> Agree, it's part of code from 2020 to fix dark theme on Android.

I'll clean it up.

**Comment by blckmn** [comment]
@haslinghuis @nerdCopter this is good to go as is... I will fix up the call backs separately in another PR

**Review by haslinghuis — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**
